File: Lab1/Polyalphabetic Cipher/poly.py
def skibidi(plaintext, keyword):
    secret_text_hehe = ""
    keyword_index = 0
    keyword_length = len(keyword)

    for character in plaintext:
        if character.isalpha():  # only process alphabetic characters
            allthewayup = character.isupper()  # check if it's uppercase
            offset = ord('A') if allthewayup else ord('a')
            key_shift = ord(keyword[keyword_index % keyword_length].lower()) - ord('a')  # calculate shift
            secret_text_hehe += chr((ord(character) - offset + key_shift) % 26 + offset)
            keyword_index += 1
        else:
            # just append non-alphabetic characters
            secret_text_hehe += character

    return secret_text_hehe


# get inputs from user
plaintext = input("enter plaintext: ")
keyword = input("enter key: ")

# encrypt and print the result
result = skibidi(plaintext, keyword)
print(f"encrypted text: {result}")

File: Lab1/Polyalphabetic Cipher/poly.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void polya(char plaintext[], char key[]) {
    int textLength = strlen(plaintext);
    int keyLength = strlen(key);
    char encryptedText[100];
    int keyIndex = 0;

    for (int i = 0; i < textLength; i++) {
        char currentChar = plaintext[i];

        // check if the character is alphabetic
        if (isalpha(currentChar)) {
            char offset = isupper(currentChar) ? 'A' : 'a'; // check for uppercase or lowercase
            int keyShift = tolower(key[keyIndex % keyLength]) - 'a'; // get shift value from key
            encryptedText[i] = ((currentChar - offset + keyShift) % 26) + offset;
            keyIndex++;
        } else {
            // if not alphabetic, it is what t is
            encryptedText[i] = currentChar;
        }
    }

    encryptedText[textLength] = '\0'; // null-terminate  string
    printf("encrypted text: %s\n", encryptedText);
}

int main() {
    char plaintext[100];
    char key[100];

    printf("enter plaintext: ");
    fgets(plaintext, sizeof(plaintext), stdin);
    plaintext[strcspn(plaintext, "\n")] = '\0'; // newline rem garcha

    printf("enter key: ");
    fgets(key, sizeof(key), stdin);
    key[strcspn(key, "\n")] = '\0'; // newline remove garcha

    polya(plaintext, key);

    return 0;
}
